src/restconfig/src/test/java/org/geoserver/rest/security/UrlCheckControllerTest.java

```diff
@@ -0,0 +1,375 @@
+/* (c) 2024 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.rest.security;
+
+import static org.custommonkey.xmlunit.XMLAssert.assertXpathEvaluatesTo;
+import static org.geoserver.rest.RestBaseController.ROOT_PATH;
+import static org.junit.Assert.*;
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+
+import java.util.List;
+import net.sf.json.JSON;
+import net.sf.json.JSONArray;
+import net.sf.json.JSONObject;
+import org.geoserver.platform.GeoServerExtensions;
+import org.geoserver.security.urlchecks.AbstractURLCheck;
+import org.geoserver.security.urlchecks.RegexURLCheck;
+import org.geoserver.security.urlchecks.URLCheckDAO;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.w3c.dom.Document;
+
+public class UrlCheckControllerTest extends SecurityRESTTestSupport {
+
+    private URLCheckDAO urlCheckDao;
+
+    @Before
+    public void setUp() throws Exception {
+
+        urlCheckDao = GeoServerExtensions.bean(URLCheckDAO.class, applicationContext);
+
+        RegexURLCheck check1 = new RegexURLCheck("check1", "regex check 1", "check1://.*");
+        RegexURLCheck check2 = new RegexURLCheck("check2", "regex check 2", "check2://.*");
+        RegexURLCheck check3 = new RegexURLCheck("check3", "regex check 3", "check3://.*");
+
+        check2.setEnabled(false);
+
+        urlCheckDao.saveChecks(List.of(check1, check2, check3));
+    }
+
+    @Test
+    public void testGetAllAsXml() throws Exception {
+
+        Document dom = getAsDOM(ROOT_PATH + "/urlchecks.xml");
+
+        assertXpathEvaluatesTo("3", "count(//urlCheck)", dom);
+
+        assertXpathEvaluatesTo("check1", "//urlCheck[1]/name", dom);
+        assertXpathEvaluatesTo(
```

Comment:
Can you validate also the contents of at least one URl check? (I expect to see that the element name is RegexURLCheck, and that it has a regex field).